addons/binding/org.openhab.binding.leapmotion/src/main/java/org/openhab/binding/leapmotion/handler/LeapMotionHandler.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.leapmotion.handler;
+
+import static org.openhab.binding.leapmotion.LeapMotionBindingConstants.*;
+
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.leapmotion.leap.CircleGesture;
+import com.leapmotion.leap.Controller;
+import com.leapmotion.leap.FingerList;
+import com.leapmotion.leap.Frame;
+import com.leapmotion.leap.Gesture;
+import com.leapmotion.leap.GestureList;
+import com.leapmotion.leap.Hand;
+import com.leapmotion.leap.Listener;
+
+/**
+ * The {@link LeapMotionHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Kai Kreuzer - Initial contribution
+ * @author Thomas Eichstädt-Engelen - Initial version of listener logic
+ *
+ */
+public class LeapMotionHandler extends BaseThingHandler {
```

Comment:
This is the only class that did not get the `@NonNullByDefault` treatment against NPEs. ;-)

Reply:
Yeah, the skeleton didn't create that at the time that I used it.
I've adapted the code now.